xdsl/irdl/declarative_assembly_format.py

```diff
@@ -11,6 +11,7 @@
 from dataclasses import dataclass, field
 from typing import Any, Literal, cast
 
+from xdsl.dialects.builtin import UnitAttr
```

Comment:
this seems unfortunate, I would like to avoid importing any dialects here, it seems like it breaks the layering conceptually

Reply:
Can we live with the fact that it would print an extra whitespace?

Reply:
Where is the whitespace printed?

Reply:
> this seems unfortunate, I would like to avoid importing any dialects here, it seems like it breaks the layering conceptually

I don't think we could avoid it *and* give special treatment to those here, which is necessary for compatibility

Reply:
But I don't understand the special treatment. Could you maybe post what happens today without these changes?

Reply:
Ah I see it was added in the latest commit. Which is the line of code that prints the extra space? Shouldn't it be avoided? Do we want this to work differently for UnitAttr to any other parametrized attribute with no parameters?

Reply:
I think the support for UnitAttr should be structured differently too; I think a custom directive would make more sense, rather than adapting the current one to special case UnitAttr.

Reply:
More specifically, I think that rather than hacking UnitAttr, the generic name-eliding feature and all to behave syntactically like we want, it would be way clearer to have a custom directive (maybe an OptionaVariable subclass) encoding the corner case of UnitAttr.
UnitAttr in declarative syntax is a corner-case by itself, and I think it would be clearer this way.
https://mlir.llvm.org/docs/DefiningDialects/Operations/#unit-attributes

Reply:
I'm happy to have a short call to clarify if it's unclear - I know that part of the code base can be a bit overwhelming :upside_down_face:

Reply:
No time for calls today, let's just keep iterating here

Reply:
Change of direction done!